numpy/tests/test_public_api.py

```diff
@@ -471,3 +472,18 @@ def check_importable(module_name):
         raise AssertionError("Modules that are not really public but looked "
                              "public and can not be imported: "
                              "{}".format(module_names))
+
+if sys.flags.optimize < 2:
```

Comment:
The normal way to write this would be to decorate the `test_` function with `@pytest.mark.skipif(sys.flags.optimize < 2)`